### DISCUSSION TITLE
UE4SS UE features to be added

### DISCUSSION THREAD

**Discussion by narknon** [UE4SS-RE/RE-UE4SS#189] [Ideas]
Please leave comments on this topic for any additional classes/types for which you would like support to be added.

Additional classes, etc., to dump with UVTD:
UDatatable
FConsoleManager
IConsoleManager

Types to add manually to UE4SS and to add to member var types that can be dumped:
MapFlags
SetFlags
EArrayPropertyFlags

Additional UE types to support/implement
Finish TMap support
TBitArray
TSparseArray
TSet
TMapBase

**Comment by Cranch-fur** [comment]
Unreal Engine LOG resurrection / hooking. Up to this day, when Unreal Engine is compiled with `#def NO_LOGGING` definition, while most of the functions are dummied out
![image](https://github.com/user-attachments/assets/a430fdfd-fd7e-464e-b85d-e9a3a9c8bba9)
![image](https://github.com/user-attachments/assets/30b639be-4172-4441-ac16-3da8a82ac84e)

`FOutputDevice::LogfImpl` seem to always be in tact no matter what
![image](https://github.com/user-attachments/assets/7454cc0f-75e8-4513-b772-dfc6aab4a5ee)
![image](https://github.com/user-attachments/assets/a5a09ee7-d487-4c72-a79e-22a5378cd9d7)

Would it make sense to resurrect the LOGS on Shipping & QA? Would it be possible to either redirect all of the dummy functions calls to operational one OR hook dummy functions execution at runtime, so end-user has an access to the game LOG?

**Reply by Cranch-fur** [reply]
Having an access to the Engine LOGS would allow to easily detect issues of all kinds from within models, animation assets, sound assets, Blueprints, and much more...
